src/shared/Core/ApplicationBase.cs

```diff
@@ -75,6 +77,31 @@ public Task<int> RunAsync(string[] args)
                 Context.Trace.WriteLine("Tracing of secrets is enabled. Trace output may contain sensitive information.");
             }
 
+            // Enable TRACE2 tracing
+            if (Context.Settings.GetTrace2TracingEnabled(out string trace2Value))
+            {
+                var connectionPath = PlatformUtils.IsWindows() ? trace2Value.TrimUntilLastIndexOf("/") :
+                    trace2Value.TrimUntilIndexOf(":");
+
+                if (!string.IsNullOrEmpty(connectionPath))
+                {
+                    Context.Trace2.AddListener(new CollectorWriter(
+                            () => new NamedPipeClientStream(".", connectionPath, PipeDirection.Out,
+                                PipeOptions.Asynchronous)
+                        )
+                    );
+
+                    if (AssemblyUtils.TryGetAssemblyVersion(out string version))
+                    {
+                        Context.Trace2.WriteVersion(version);
```

Comment:
The version event is required.  It sets the absolute wall-clock of the start of the program and causes other things to be initialized in the daemon.

Could we change TryGetAssemblyVersion() always return a version number value -- even if something like "0.0" or "0.0.0".

Reply:
I'm hesitant to do this, since other code relies on `TryGetAssemblyVersion` working as it does. Perhaps we could check to see if the result is false and then add a "dummy" version here based on that. @mjcheetham - any thoughts? I'm definitely open to a cleaner solution.

Reply:
That's fine.  Let it get the real version number from the assembly (which will probably be updated by the CI/Release build scripts), and fallback to a constant that you hard-code here if the assembly function fails.

Reply:
(It's really not that big a deal.)